.vscode/launch.json

```diff
@@ -59,6 +59,15 @@
           "remoteRoot": "/usr/src/homeassistant"
         }
       ]
+    },
+    {
+      "name": "Home Assistant: Debug Tests",
+      "type": "python",
+      "request": "launch",
+      "program": "${file}",
```

Comment:
How would this work?

Reply:
> Do you mean "program: "S{file}"?

Yup, comment was on that line :)

> If that's what you mean then I don't know the details as it's internal to the vscode test framework.

Right, hence my question, as this looks like a copy/paste from docs, without knowing what it does. That is a red flag for me.

> You don't actually launch this debug configuration directly

Well adding it here, means you can. That is what I find odd about it.

> I use test explorer and it also picks up the setting if present.

We have disabled that one by default, as our test suite is too large for it to make it work correctly.

../Frenck

Reply:
> > If that's what you mean then I don't know the details as it's internal to the vscode test framework.
> 
> Right, hence my question, as this looks like a copy/paste from docs, without knowing what it does. That is a red flag for me.
> 

I'm comfortable I know what it does because the official documentation is clear and I know that without it I can't step into library code and I get this message from the test framework:

```
Frame skipped from debugging during step-in.
Note: may have been skipped because of "justMyCode" option (default == true). Try setting "justMyCode": false in the debug configuration (e.g., launch.json).
```
Whereas with this configuration it steps in just fine.

> > You don't actually launch this debug configuration directly
> 
> Well adding it here, means you can. That is what I find odd about it.
> 

That's true, it would be good if vscode hid this config from the run drop down. I just tried to run it directly and it does indeed try to run the tests in the current file.  N.B. I have also tried adding it to a launch section of my user settings.json but it is not read from there.

> > I use test explorer and it also picks up the setting if present.
> 
> We have disabled that one by default, as our test suite is too large for it to make it work correctly.

I have test explorer working ok by restricting my tests to specific integrations in settings.json with `"python.testing.pytestArgs"`. To be honest I have wondered how other developers are dealing with debugging tests in vscode in general.

Anyway I thought this PR could be helpful for others who may have had the same problems as me, and justMyCode is false for the other configurations so it seemed consistent. Happy to close if you think it could cause more harm than good.

> ../Frenck

Reply:
So what do you think about my comment above @frenck?